### PR TITLE
Enable Windows CI

### PR DESCRIPTION
**Pull request by liufengyun**
Enable run and pos tests on windows

### CODE REVIEW

File: compiler/test/dotty/tools/vulpix/ParallelTesting.scala [hunk 1]
```diff
@@ -606,7 +606,12 @@ trait ParallelTesting extends RunnerOrchestration { self =>
       testSource.sourceFiles.foreach { file =>
         if checkFiles.contains(file) then
           val checkFile = checkFiles(file)
-          val actual = Source.fromFile(file, "UTF-8").getLines().toList
+          val actual = {
+            val source = Source.fromFile(file, "UTF-8")
+            val lines  = source.getLines().toList
+            source.close()
+            lines
```
Comment (sjrd) [inline compiler/test/dotty/tools/vulpix/ParallelTesting.scala:613]: Consider using a `try..finally` here to close the source:
```suggestion
            val source = Source.fromFile(file, "UTF-8")
            try source.getLines().toList
            finally source.close()
```

**Review by sjrd — APPROVED** [review]
Thank you so much @liufengyun for making this work on Windows!

**Review by sjrd — APPROVED** [review]
Thank you so much @liufengyun for making this work on Windows!

**Comment by odersky** [comment]
It seems the CI is down or hangs somehwere. All open PRs I see are queued and none is making progress.

**Comment by liufengyun** [comment]
> It seems the CI is down or hangs somehwere. All open PRs I see are queued and none is making progress.

It is still running, just that there is only one worker, thus moves slowly.

After everything is green, I'll investigate how to move Windows CI from `appveyor` to Github actions and have multiple workers.

File: compiler/src/dotty/tools/dotc/util/SourceFile.scala [hunk 2]
```diff
@@ -212,7 +212,7 @@ object SourceFile {
   def fromId(id: Int): SourceFile = sourceOfChunk(id >> ChunkSizeLog)
 
   def virtual(name: String, content: String, maybeIncomplete: Boolean = false) =
-    val src = new SourceFile(new VirtualFile(name, content.getBytes), scala.io.Codec.UTF8)
+    val src = new SourceFile(new VirtualFile(name, content.getBytes("UTF-8")), scala.io.Codec.UTF8)
```
[the remark this answers is not in the document]
Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/util/SourceFile.scala]: Now it's done for the whole repo.

**Comment by liufengyun** [comment]
Now both `sbt test` and `sbt bootstrapped/test` should succeed on Windows.

- `sbt test`
    It has been green on appveyor (but it's unstable)
- `sbt bootstrapped/test`
    I haven't be able to test it on the Windows virtual machine due to memory quota. The 3 language server test cases that fail are disabled on Windows.

However, **we should not merge this PR** until we migrate to a infrastructure that can run the jobs _fast_ & _reliably_.

**Comment by sjrd** [comment]
`dotty-bootstrapped` completed in 16 minutes on my Windows machine, but some tests failed. The full transcript is in the following gist: https://gist.github.com/sjrd/cdb570d5a3fef11edc5dcb013a5e3df4
FTR that sbt instance gets allocated about 1GB of RAM.

**Comment by liufengyun** [comment]
Thanks a lot @sjrd !

**Comment by liufengyun** [comment]
@sjrd : what about we merge the PR, and figure out the infrastructure in another PR?

**Comment by sjrd** [comment]
Yes, I think we can merge it as is. At least now `test` and `dotty-bootstrap/test` pass on my Windows machine, which is a huge improvement over the status quo.

**Pull request merged**